### PR TITLE
Fix 9814 file system paths "pragma: \ define" tiddler

### LINKED ISSUE #9814: [Report] FileSystemPaths filter creates wrong directory for "Pragma: \define" tiddler
Tiddler `Pragma: \define` (tagged `Pragmas`) is matched by filter `[tag[Pragmas]addprefix[pragmas/]]` in `$:/config/FileSystemPaths`.

Actual behaviour on Windows: creates dir `tiddlers/pragmas/Pragma_ /` with file `define.tid`.
Expected: `tiddlers/pragmas/Pragma_ _define.tid`.

This is caused by `$tw.utils.generateTiddlerFilepath` (core-server/filesystem.js:356) does not sanitise `\` in its cross-platform-filename regex.

PR with new tests on the way.

### PR DESCRIPTION
**Pull request by pmario**
This PR fixes #9814 

- #9814 

This PR contains the fix and some new tests for all the forbidden characters.

### CODE REVIEW

**Comment by pmario** [comment]
This fix it too aggressive if `root/community/` tiddlers are touched using `$:/config/FileSystemPaths` filters that have eg: `[tag[Community]...]`. Where tag: Community is also shared between `root/community` tiddlers and `edition/tw5.com/tiddlers/`

**TODO**: Much better tests for edge cases needed

**Comment by pmario** [comment]
@Jermolene 

We should discuss if we move the `root/community` and other "out of bounds" tiddlers back to well established directory structure eg: `/editions/community/tiddlers` where they also can be easily be imported into different other editions if we need to

IMO Those experimental directories only cause problems.

File: core-server/filesystem.js [hunk 1]
```diff
@@ -352,8 +364,12 @@ exports.generateTiddlerFilepath = function(title,options) {
 	}
 	// Replace any Unicode control codes
 	filepath = filepath.replace(/[\x00-\x1f\x80-\x9f]/g,"_");
-	// Replace any characters that can't be used in cross-platform filenames
-	filepath = $tw.utils.transliterate(filepath.replace(/<|>|~|\:|\"|\||\?|\*|\^/g,"_"));
+	// Replace any characters that can't be used in cross-platform filenames.
+	// Skipped for pinned paths because originalpath uses \ as legitimate
```
Comment (saqimtiaz) [inline core-server/filesystem.js]:
```suggestion
```

File: core-server/filesystem.js [hunk 2]
```diff
@@ -352,8 +364,11 @@ exports.generateTiddlerFilepath = function(title,options) {
 	}
 	// Replace any Unicode control codes
 	filepath = filepath.replace(/[\x00-\x1f\x80-\x9f]/g,"_");
-	// Replace any characters that can't be used in cross-platform filenames
-	filepath = $tw.utils.transliterate(filepath.replace(/<|>|~|\:|\"|\||\?|\*|\^/g,"_"));
+	// Replace any characters that can't be used in cross-platform filenames.
+	// path separator on Windows.
```
Comment (saqimtiaz) [inline core-server/filesystem.js]:
```suggestion
```

**Comment by saqimtiaz** [comment]
Thank you @pmario
